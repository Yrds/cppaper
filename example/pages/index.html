<!DOCTYPE html>
<html>
  <head>
    <meta charset="UTF-8" />
    <meta name="viewport" content="width=device-width" />
    <title> {{page.title}} </title>
  </head>
  <body>
    {% for page in getPagesFrom(".") %}
      <a href="{{page.path}}">{{page.title}}</a>

      {% set config=getConfigFrom(page.id) %}


      <!-- 
        {{config}}
      -->
    {% endfor %}
  </body>
</html>
