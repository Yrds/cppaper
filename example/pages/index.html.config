title=Custom Title
tags=test,article
