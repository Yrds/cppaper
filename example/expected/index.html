<!DOCTYPE html>
<html>
  <head>
    <meta charset="UTF-8" />
    <meta name="viewport" content="width=device-width" />
    <title> Custom Title </title>
  </head>
  <body>
    <h2> getPagesFrom example </h2>
    <ul>
      <li>
    

      
      <!-- 
        {"domain":"example.com","lua_libraries":"base,string,os","markdown_unsafe":"true","name":"Yuri Santos","template":"cppaper-default/default.html"}
      -->
      <a href=".nojekyll.html">.nojekyll</a>

    

      
      <!-- 
        {"domain":"example.com","lua_libraries":"base,string,os","markdown_unsafe":"true","name":"Yuri Santos","template":"cppaper-default/default.html"}
      -->
      <a href="test_lua.html">test_lua</a>

    

      
      <!-- 
        {"domain":"example.com","lua_libraries":"base,string,os","markdown_unsafe":"true","name":"Yuri Santos","tags":"test","template":"cppaper-default/default.html","title":"Test Page"}
      -->
      <a href="test.html">Test Page</a>

    

      
      <!-- 
        {"domain":"example.com","lua_libraries":"base,string,os","markdown_unsafe":"true","name":"Yuri Santos","tags":"test,article","template":"cppaper-default/default.html","title":"Custom Title"}
      -->
      <a href="index.html">Custom Title</a>

    
      </li>
    </ul>

    <h2>Pages with the tag 'test'</h2>
    <ul>
      
      <li>{"domain":"example.com","index":"json-default/index.html","lua_libraries":"base,string,os","name":"Yuri Santos","tags":"test","template":"json-default/default.html"}: json-test/json_test.html</li>
      
      <li>{"domain":"example.com","lua_libraries":"base,string,os","markdown_unsafe":"true","name":"Yuri Santos","tags":"test","template":"cppaper-default/default.html","title":"Test Page"}: test.html</li>
      
      <li>{"domain":"example.com","lua_libraries":"base,string,os","markdown_unsafe":"true","name":"Yuri Santos","tags":"test,article","template":"cppaper-default/default.html","title":"Custom Title"}: index.html</li>
      
      <li></li>
    </ul>

    <h2>Pages with the config 'title'</h2>
    <ul>
      <li>{"domain":"example.com","lua_libraries":"base,string,os","markdown_unsafe":"true","name":"Yuri Santos","tags":"test,article","template":"cppaper-default/default.html","title":"Custom Title"}: index.html</li>
      <li>{"domain":"example.com","lua_libraries":"base,string,os","markdown_unsafe":"true","name":"Yuri Santos","tags":"test","template":"cppaper-default/default.html","title":"Test Page"}: test.html</li>
      <li></li>
    </ul>

    <h2>Pages with the config title = 'Custom Title'</h2>
    <ul>
      <li>{"domain":"example.com","lua_libraries":"base,string,os","markdown_unsafe":"true","name":"Yuri Santos","tags":"test,article","template":"cppaper-default/default.html","title":"Custom Title"}: index.html</li>
      <li></li>
    </ul>

    <h2>Pages with the config markdown_unsafe = 'true'</h2>
    <ul>
      <li>{"domain":"example.com","lua_libraries":"base,string,os","markdown_unsafe":"true","name":"Yuri Santos","tags":"test,article","template":"cppaper-default/default.html","title":"Custom Title"}: index.html</li>
      <li>{"domain":"example.com","lua_libraries":"base,string,os","markdown_unsafe":"true","name":"Yuri Santos","tags":"test","template":"cppaper-default/default.html","title":"Test Page"}: test.html</li>
      <li>{"domain":"example.com","lua_libraries":"base,string,os","markdown_unsafe":"true","name":"Yuri Santos","template":"cppaper-default/default.html"}: test_lua.html</li>
      <li>{"domain":"example.com","lua_libraries":"base,string,os","markdown_unsafe":"true","name":"Yuri Santos","no_output":"true","template":"cppaper-default/default.html"}: nooutput.html</li>
      <li>{"domain":"example.com","lua_libraries":"base,string,os","markdown_unsafe":"true","name":"Yuri Santos","template":"cppaper-default/default.html"}: .nojekyll</li>
      <li></li>
    </ul>

  </body>
</html>
